elastic/fleet-server
# Add prometheus metrics endpoint

## What is the problem this PR solves?

Gives a try to exposing Fleet Server metrics through prometheus.

## How does this PR solve the problem?

* Register a prometheus handler, under the metrics endpoint in the `/metrics` path.
* Register an "info" metric with the service information.
* Adds wrappers that include prometheus and beats metrics, methods exposed by the wrapper modifies both wrapped metrics.

## How to test this PR locally

* Run fleet-server.
* Check that metrics under the `/metrics` path include at least the same metrics as the beats metrics.

## Checklist





## Related issues


- Fixes https://github.com/elastic/fleet-server/issues/2542
